### PR TITLE
Fixs #30324 - check for http(s) proxy

### PR DESCRIPTION
**Pull request by sbernhard**

### CODE REVIEW

**Comment by kgaikwad** [comment]
Thank you @sbernhard for contribution. 

I would like to know that are you running this check in any external script before `installer` run? OR
do you want to run this check as a part of any specific scenario like upgrade?

**Comment by sbernhard** [comment]
I would run it before every task which uses foreman-installer, like upgrades.

File: definitions/checks/env_proxy.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Checks::EnvProxy < ForemanMaintain::Check
+  metadata do
+    label :env_proxy
+    tags :env_proxy
+    description 'Check if proxy is set in ENV which shouldn\'t be the case'
```
Comment (kgaikwad) [inline definitions/checks/env_proxy.rb:5]:
```suggestion
    description 'Check to make sure no HTTP(S) proxy set in ENV'
```

**Comment by kgaikwad** [comment]
Thank you @sbernhard!

**Pull request merged**